org.eclipse.winery.edmm/src/main/java/org/eclipse/winery/edmm/utils/ZipUtility.java

```diff
@@ -0,0 +1,88 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+public class ZipUtility {
+
+    /**
+     * The function is used to zip a directory in a file
+     * @param sourcePath the path of the directory
+     * @param destPath the destination path of the zip file,
+     *                 if a zip file already exists at the destination path, it will be overwritten
+     *
+     * https://stackoverflow.com/questions/15968883/how-to-zip-a-folder-itself-using-java
+     */
+    public static ZipOutputStream pack(Path sourcePath, Path destPath) throws IOException {
+        if (Files.exists(destPath)) {
+            Files.delete(destPath);  
+        }
+        Files.createFile(destPath);
+        
+        ZipOutputStream zipOutputStream = new ZipOutputStream(Files.newOutputStream(destPath));
+        try {
+            Files.walk(sourcePath)
+                .filter(path -> !Files.isDirectory(path))
+                .forEach(path -> {
+                    ZipEntry zipEntry = new ZipEntry(sourcePath.relativize(path).toString());
+                    try {
+                        zipOutputStream.putNextEntry(zipEntry);
+                        Files.copy(path, zipOutputStream);
+                        zipOutputStream.closeEntry();
+                    } catch (IOException e) {
+                        System.err.println(e);
+                    }
+                });
+        } catch (IOException e){
+            throw new IOException();
+        }
+        return zipOutputStream;
+    }
+
+    /**
+     * 
+     * @param zipFilePath the Path of the zip file to unpack
+     * @param destDirPath the Path of the directory where we want the unzipped files, 
+     *                    if the direcory exists it will be overwritten
+     * @return it return destDirPath
+     * 
+     * https://www.baeldung.com/java-compress-and-uncompress
+     */
+    public static Path unpack(Path zipFilePath, Path destDirPath) throws IOException{
+        if (Files.exists(destDirPath)) {
+            Files.delete(destDirPath);
+        }
+        Files.createDirectory(destDirPath);
+        
+        byte[] buffer = new byte[1024];
+        ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(zipFilePath.toFile()));
+        
+        ZipEntry zipEntry = zipInputStream.getNextEntry();
+        while (zipEntry != null) {
+            
+            File newFile = new File(destDirPath.toFile(), zipEntry.getName());
+            
+            // creating the directories in which the newFile shoud be stored
+            newFile.toPath().getParent().toFile().mkdirs();
+                        
+            FileOutputStream fos = new FileOutputStream(newFile);
+            int len;
+            while ((len = zipInputStream.read(buffer)) > 0) {
+                fos.write(buffer, 0, len);
+            }
+            fos.close();
```

Comment:
This should be replaceable by using 

```suggestion
            Path newFile = destDirPath.resolve(zipEntry.getName());
            Files.createDirectories(newFile.getParent());
            Files.copy(zipInputStream, newFile);
```

Reply:
Yes, your code is better. It can be committed I think..